sway-core/src/sway.pest

```diff
@@ -41,32 +41,39 @@ predicate    =  {"predicate" ~ ";" ~ (non_var_decl|use_statement|include_stateme
 file_path         = { ident ~ ("/" ~ ident)* }
 include_statement = { include_keyword ~ file_path ~ alias? ~ ";"}
 alias             = { as_keyword ~ ident }
+
+// this call path is used when there can be some number of "::foo"s
+call_path           =  {relative_call_path|absolute_call_path}
+relative_call_path  =  {ident ~ (path_separator ~ ident)*}
+absolute_call_path  =  {path_separator ~ ident ~ (path_separator ~ ident)*}
+// this call path is used when there must be at least one "::foo"
+call_path_          =  {relative_call_path_|absolute_call_path_}
```

Comment:
can you comment the differences between the ones with an underscore suffixing them and those that don't have that in a bit more detail? it isn't immediately obvious to me